rewrite-java/src/main/java/org/openrewrite/java/RemoveObjectsIsNull.java

```diff
@@ -65,10 +66,19 @@ private Expression replace(ExecutionContext ctx, J.MethodInvocation m, String pa
                 }
 
                 // Replace the method invocation with a simple null check
-                Expression e = m.getArguments().get(0);
-                Expression replaced = JavaTemplate.apply(pattern, getCursor(), m.getCoordinates().replace(), e);
+                getCursor().getParentTreeCursor().putMessage("SIMPLIFY_BOOLEAN_EXPRESSION", true);
+                Expression replaced = JavaTemplate.apply(pattern, getCursor(), m.getCoordinates().replace(), m.getArguments().get(0));
                 return (Expression) new UnnecessaryParenthesesVisitor<>().visitNonNull(replaced, ctx, getCursor());
             }
+
+            @Override
+            public J postVisit(J tree, ExecutionContext ctx) {
+                J j = super.postVisit(tree, ctx);
+                if (getCursor().getNearestMessage("SIMPLIFY_BOOLEAN_EXPRESSION", false)) {
```

Comment:
I am AFK, but I would expect `pollMessage()` rather than `getNearestMessage()`. I doubt that will fix the problem, but that should be better.